IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -469,54 +532,165 @@ - (MSIDCredentialCacheItem *)tokenWithKey:(__unused MSIDCacheKey *)key
-- (BOOL)removeItemsWithTokenKey:(__unused MSIDCacheKey *)key
-                        context:(__unused id<MSIDRequestContext>)context
-                          error:(__unused NSError **)error
+- (BOOL)removeItemsWithTokenKey:(MSIDCacheKey *)key
+                        context:(id<MSIDRequestContext>)context
+                          error:(NSError **)error
 {
-    return [self removeItemsWithKey:key context:context error:error];
+    MSID_TRACE;
+    NSString *account = key.account;
+    NSString *service = key.service;
+    
+    MSID_LOG_VERBOSE(context, @"Remove keychain items, key info (account: %@ service: %@, keychainGroup: %@)", _PII_NULLIFY(account), _PII_NULLIFY(service), [self keychainGroupLoggingName]);
+    MSID_LOG_VERBOSE_PII(context, @"Remove keychain items, key info (account: %@ service: %@, keychainGroup: %@)", account, service, self.keychainGroup);
+    
+    if (!key || !(key.service || key.account))
+    {
+        MSID_LOG_ERROR(context, @"Key is nil or one of the key attributes account or service is nil.");
+        [self createError:@"Key is nil or one of the key attributes account or service is nil."
+                   domain:MSIDErrorDomain errorCode:MSIDErrorInvalidDeveloperParameter error:error context:context];
+        return NO;
+    }
+    
+    if (key.isShared)
+    {
+        return [self removeSharedCredentialsWithTokenKey:key context:context error:error];
+    }
+    else
+    {
+        return [self removeAppCredentialsWithTokenKey:key context:context error:error];
+    }
+}
+
+- (BOOL)removeSharedCredentialsWithTokenKey:(MSIDCacheKey *)key
+                                    context:(id<MSIDRequestContext>)context
+                                      error:(NSError **)error
+{
+    MSIDMacSharedCredentialCacheItem *sharedCredential = [self sharedCredentialWithKey:key context:context error:error];
+    
+    if (sharedCredential)
+    {
+        [sharedCredential removeSharedCredentialForKey:(MSIDDefaultCredentialCacheKey *)key];
+        return [self saveSharedCredential:sharedCredential key:key context:context error:error];
+    }
+    
+    return YES;
+}
+
+- (BOOL)removeAppCredentialsWithTokenKey:(MSIDCacheKey *)key
+                                 context:(id<MSIDRequestContext>)context
+                                   error:(NSError **)error
+{
+    MSIDMacAppCredentialCacheItem *appCredential = [self appCredentialWithKey:key context:context error:error];
+    
+    if (appCredential)
+    {
+        [appCredential removeAppCredentialForKey:(MSIDDefaultCredentialCacheKey *)key];
+        return [self saveAppCredential:appCredential key:key context:context error:error];
+    }
+    
+    return YES;
+}
+
+- (MSIDMacSharedCredentialCacheItem *)sharedCredentialWithKey:(MSIDCacheKey *)key
+                                                      context:(id<MSIDRequestContext>)context
+                                                        error:(NSError **)error
```

Comment:
The sharedCredentialWithKey  and appCredentialWithKey methods don't actually use the passed-in key, which seems strange given the method's names. If this and the error parameter aren't actually being used, they should have an __unused flag.

Reply:
Removed both methods in the new commit.